src/org/tensorflow/yolo/view/ClassifierActivity.java

```diff
@@ -136,11 +184,65 @@ private void renderAdditionalInformation(final Canvas canvas) {
             }
         }
 
-        lines.add("Frame: " + previewWidth + "x" + previewHeight);
-        lines.add("View: " + canvas.getWidth() + "x" + canvas.getHeight());
-        lines.add("Rotation: " + sensorOrientation);
-        lines.add("Inference time: " + lastProcessingTimeMs + "ms");
+// shimatani
+        // lines.add("Frame: " + previewWidth + "x" + previewHeight);
+        // lines.add("View: " + canvas.getWidth() + "x" + canvas.getHeight());
+        // lines.add("Rotation: " + sensorOrientation);
+        // lines.add("Inference time: " + lastProcessingTimeMs + "ms");
+
+        Log.d(LOGGING_TAG, String.format("検出名: %s", nowRecognizedClass));
+        lines.add("検出名: " + nowRecognizedClass);
+
+        String msgInf = makeTts(nowRecognizedClass);
+
+        lines.add(msgInf1);
+        lines.add(msgInf2);
 
         borderedText.drawLines(canvas, 10, 10, lines);
     }
+
+    private void run() {
+        final long startTime = SystemClock.uptimeMillis();
+        final List<Recognition> results = recognizer.recognizeImage(croppedBitmap);
+        lastProcessingTimeMs = SystemClock.uptimeMillis() - startTime;
+        overlayView.setResults(results);
+
+        // shimatani
+        Log.d(LOGGING_TAG, "Debug: runInBackground()");
+
+        if (!(results.isEmpty())) {
+            nowResult = results.get(0).getTitle();
+            Log.d(LOGGING_TAG, String.format("Find: %s", nowResult));
+
+            if (!(results.isEmpty()) && lastResult.equals(nowResult)) {
+                matchCount++;
+                if (matchCount > 3) {
```

Comment:
It looks for me more like a workaround to have a threshold above the recognized classes. The main problem is either the model is not well trained thus it often gives you bad results or the threshold is not properly set in your configuration file. 
Another question is why you chose the 3 magic number as comparison value? 
Unfortunately, I cannot consider these changes as a good solution for the real problem.

Reply:
As you say, I think this is not an essential improvement.

I think that it is originally to make a model with high accuracy, but I have taken easier improvement measures than improving the accuracy of the model.

During operation I noticed that this application was immediately judging if it detected an object even before holding it over the object.
Therefore, I tried to judge it as an object (speaking) when it matched four times (the value for comparison is 3).

I will reselect the images for learning and try to improve the accuracy of the model. Please let me know if you know any other effective method.